light-client/src/supervisor.rs

```diff
@@ -338,50 +388,23 @@ impl Handle {
         receiver.recv().unwrap()
     }
 
-    /// Verify to the highest block.
-    pub fn verify_to_highest(&mut self) -> Result<LightBlock, Error> {
+    fn verify_to_highest(&mut self) -> Result<LightBlock, Error> {
         self.verify(HandleInput::VerifyToHighest)
     }
 
-    /// Verify to the block at the given height.
-    pub fn verify_to_target(&mut self, height: Height) -> Result<LightBlock, Error> {
+    fn verify_to_target(&mut self, height: Height) -> Result<LightBlock, Error> {
         self.verify(|callback| HandleInput::VerifyToTarget(height, callback))
     }
 
-    /// Verify either to the latest block (if `height == None`) or to a given block (if `height == Some(height)`).
-    fn verify(
-        &mut self,
-        make_event: impl FnOnce(Callback<Result<LightBlock, Error>>) -> HandleInput,
-    ) -> Result<LightBlock, Error> {
-        let (sender, receiver) = channel::bounded::<Result<LightBlock, Error>>(1);
-
-        let callback = Callback::new(move |result| {
-            sender.send(result).unwrap();
-        });
-
-        let event = make_event(callback);
-        self.sender.send(event).unwrap();
-
-        receiver.recv().unwrap()
-    }
-
-    /// Async version of `verify_to_highest`.
-    ///
-    /// The given `callback` will be called asynchronously with the
-    /// verification result.
-    pub fn verify_to_highest_async(
+    fn verify_to_highest_async(
         &mut self,
         callback: impl FnOnce(Result<LightBlock, Error>) -> () + Send + 'static,
     ) {
         let event = HandleInput::VerifyToHighest(Callback::new(callback));
         self.sender.send(event).unwrap();
     }
 
-    /// Async version of `verify_to_target`.
-    ///
-    /// The given `callback` will be called asynchronously with the
-    /// verification result.
-    pub fn verify_to_target_async(
+    fn verify_to_target_async(
```

Comment:
Should we continue to include the inline docs here?

Reply:
The trait methods are documented for that reason. For doc generation purposes developers will always see the docs from the impled trait. Keeping this in two places, just lends itself to divergence and unnecessary overhead.